parquet-column/src/main/java/org/apache/parquet/column/ColumnDescriptor.java

```diff
@@ -90,14 +104,21 @@ public int getMaxDefinitionLevel() {
    * @return the type of that column
    */
   public PrimitiveTypeName getType() {
-    return type;
+    return type.getPrimitiveTypeName();
   }
 
   /**
    * @return the size of the type
    **/
   public int getTypeLength() {
-    return typeLength;
+    return type.getTypeLength();
+  }
+
+  /**
+   * Returns the full type object of the column
+   */
+  public PrimitiveType getFullType() {
```

Comment:
We should leave a comment for 2.0.0 API changes to rename this to `getType()`

Reply:
Shall we deprecate the existing `getType()` for now?

Reply:
After discussing it locally, I don't think it is a good idea to keep `getType()` for 2.0.0 and return with the `Type` object because we cannot handle deprecation properly. 
I think, it would be better to rename `getFullType()` to `getPrimitiveType()` and keep for 2.0.0 and simply remove `getType()`.